scipy/stats/_quantile.py

```diff
@@ -14,16 +14,20 @@
 from scipy.stats._axis_nan_policy import _broadcast_arrays, _contains_nan
 
 
-def _quantile_iv(x, p, method, axis, nan_policy, keepdims):
-    xp = array_namespace(x, p)
+def _quantile_iv(x, p, method, axis, nan_policy, keepdims, weights):
+    xp = array_namespace(x, p, weights)
 
     if not xp.isdtype(xp.asarray(x).dtype, ('integral', 'real floating')):
         raise ValueError("`x` must have real dtype.")
 
     if not xp.isdtype(xp.asarray(p).dtype, 'real floating'):
         raise ValueError("`p` must have real floating dtype.")
 
-    x, p = xp_promote(x, p, force_floating=True, xp=xp)
+    if not (weights is None
+            or xp.isdtype(xp.asarray(weights).dtype, ('integral', 'real floating'))):
+        raise ValueError("`weights` must have real dtype.")
+
+    x, p, weights = xp_promote(x, p, weights, force_floating=True, xp=xp)
     p = xp.asarray(p, device=xp_device(x))
```

Comment:
Now we call `xp_promote` which in turn calls `xp.asarray` is this line still needed?